### DISCUSSION TITLE
Xray(E+F+H+A)+Nginx

### DISCUSSION THREAD

**Discussion by flashcharge99** [lxhao61/integrated-examples#212] [Q&A] [answered]
for Xray(E+F+H+A)+Nginx configurations the domains can't be the same? must be different.

**Comment by lxhao61** [comment, marked as answer]
是的，SNI 分流必须域名不同才行；但可证书相同，如所有域名使用通配符证书或 SAN 证书。

**Reply by flashcharge99** [reply]
using acme.sh wildcard dns cf is it possible?

**Reply by lxhao61** [reply]
当然可以。

**Reply by flashcharge99** [reply]
I'm using nginx version 1.27.0 there is an error "a duplicate listen 127.0.0.1:7443"

**Reply by lxhao61** [reply]
见注释！根据你的 Nginx 版本删除不符合版本的配置，留下如下：
```
........
    server {
        listen 127.0.0.1:88 proxy_protocol;
        http2 on;
........
    server {
        listen 127.0.0.1:7443 ssl proxy_protocol;
        http2 on;
........
```

**Reply by flashcharge99** [reply]
means this configuration is removed?

```
   server {
      listen 127.0.0.1:81 proxy_protocol;
      listen 127.0.0.1:82 http2 proxy_protocol;
```

**Reply by lxhao61** [reply]
是的，此两项配置不符合你的 Nginx 版本！见注释：``仅版本小于 v1.25.1 才配置，否则必须删除。``

**Reply by flashcharge99** [reply]
success runs well but why can't use SNI, example config like this

```
trojan://diy6443@trojan.abc.com:443?security=tls&type=tcp&sni=static-web.prod.vidiocdn.com#Xray
```

**Reply by lxhao61** [reply]
不明白你的逻辑。就 SNI 来说，不能用为非自已证书对应的域名！
